docs/source/notes/rref.rst

```diff
@@ -0,0 +1,304 @@
+Remote Reference Protocol
+=========================
+
+.. warning::
+  The :ref:`rref` API is experimental and subject to change.
+
+This note describes the design details of Remote Reference protocol and walks
+through message flows in different scenarios. Make sure you're familiar with the
+:ref:`distributed-rpc-framework` before proceeding.
+
+Background
+^^^^^^^^^^
+
+RRef stands for Remote REFerence. It is a reference of an object which locates
+on the local or a remote worker, and transparently handles reference counting
+under the hood. Conceptually, it can be considered as a distributed shared
+pointer. Applications can create an RRef by calling
+`torch.distributed.rpc.remote`. Each RRef is owned by the callee worker of the
```

Comment:
It might be helpful to link to all these functions. You can do it with `func: torch.distributed.rpc.remote`.